test/end2end.py

```diff
@@ -120,3 +120,22 @@ def test_output(self):
         with open(f_out, 'r') as output:
             text = output.read()
             self.assertTrue(regex.match(text), "Pattern '%s' found in: %s" % (regex.pattern, text))
+
+
+    def test_hanging(self):
+        """ Test --timeout option when program has no output"""
+        no_output_mpirun = '\n'.join([
+            "#!/bin/bash",
+            "sleep 4", # sleep for 4 seconds
+            "echo 'some output'",
+            "sleep 3"
+        ])
+
+        install_fake_mpirun('mpirun', self.tmpdir, txt=no_output_mpirun)
+        ec, out = run_simple("%s %s --setmpi impirun --output-check-timeout 2 hostname" % (sys.executable, self.mympiscript))
+        self.assertEqual(ec, 0, "Command exited normally: exit code %s; output: %s" % (ec, out))
+
+        regex = re.compile("WARNING: mympirun has been running for [1-9] seconds without seeing any output.")
```

Comment:
use `.*` rather than `[1-9]`, to make sure you catch any wrongly included warnings (e.g. 0 seconds)